addons/web_tour/static/src/js/tour_service.js

```diff
@@ -116,25 +117,34 @@ return session.is_bound.then(function () {
             // Either all the mutations have been ignored or one was detected as
             // tracked and will trigger a tour manager update.
             currentMutations = [];
+            mutationTimer = undefined; // a new timer can be activated to process new incoming mutations
+            nextProcessMutation = new Date().getTime() + duration;
 
             // Update the tour manager if required.
             if (hasTrackedMutation) {
                 tour_manager.update();
             }
         }
 
-        // Use a MutationObserver to detect DOM changes. When a mutation occurs,
-        // only add it to the list of mutations to process and delay the
-        // mutation processing. We have to record them all and not in a
+        // Use a MutationObserver to detect DOM changes.
+        // When mutations occur too fast, collect them until a given time has passed.
+        // Flush all mutations collected during the time interval in one go.
+        // This avoids being continuously delayed by a long incoming stream of events.
+        // We have to record them all and not in a
         // debounced way otherwise we may ignore tracked ones in a serie of
         // 10 tracked mutations followed by an untracked one. Most of them
         // will trigger a tip check anyway so, most of the time, processing the
         // first ones will be enough to ensure that a tip update has to be done.
         let mutationTimer;
+        let nextProcessMutation = 0;
         const observer = new MutationObserver(mutations => {
-            clearTimeout(mutationTimer);
             currentMutations.push(...mutations);
-            mutationTimer = setTimeout(() => _processMutations(), 750);
+            if (!mutationTimer) { // no timer already set
+                const now = new Date().getTime();
+                const delay = now >= nextProcessMutation ? duration : nextProcessMutation - now;
+                // until time is reached (or after minimal duration, if new burst)
+                mutationTimer = setTimeout(() => _processMutations(), delay);
+            }
         });
```

Comment:
I think this implementation is the same as doing this:

```javascript
const throttledProcessMutations = _.throttle(_processMutations, duration, {leading: false});
const observer = new MutationObserver(mutations => {
    currentMutations.push(...mutations);
    throttledProcessMutations();
});
```

Could you verify?

I am still unsure about only throttling with a "small" delay though, but we can test on a multi build later on to see what's what. After all, this might just make our test system more strict I guess (forcing the DOM to be stable after 750ms after any change...) but the goal is not to have a runbot red half the time in stable versions either ;)

Reply:
Equivalence confirmed, implementation adjusted

Reply:
@bso-odoo Could you review your git history? I think you squashed your implementation with the `_.throttle` in the wrong commit. Your second commit is still the old implementation

Reply:
It was actually not equivalent: throttle will not take into account the last set of throttled events - reverting the previous approach

Reply:
>  throttle will not take into account the last set of throttled events

What do you mean?

Reply:
Beyond the first throttled group, if new events come in within the time interval, they will be collected in the mutations list, but the throttled method will not trigger. If no new event comes in after the time interval, the throttled method will not be called anymore, and the collected mutations will keep waiting in their list without ever being accessed.